clients/drcachesim/scheduler/scheduler_impl.h

```diff
@@ -734,10 +752,16 @@ template <typename RecordType, typename ReaderType> class scheduler_impl_tmpl_t
     stream_status_t
     pick_next_input(output_ordinal_t output, uint64_t blocked_time);
 
-    // Helper for pick_next_input() for MAP_AS_PREVIOUSLY.
+    // Helper for pick_next_input() specialized by mode.
     // No input_info_t lock can be held on entry.
-    stream_status_t
-    pick_next_input_as_previously(output_ordinal_t output, input_ordinal_t &index);
+    virtual stream_status_t
+    pick_next_input_try(output_ordinal_t output, uint64_t blocked_time,
```

Comment:
Why the _try suffix? Because the default one returns NOT_IMPLEMENTED? Or because the implementation may sometimes not pick next input (because of the STATUS_INVALID)? If it's just the former, then I feel the _try isn't warranted.

Reply:
If it's the latter, then add to documentation that the implementation may choose to not pick next input and in that case should return `<error code>`.

Reply:
It was only because we could loop if the chosen input is eof: but _try doesn't seem a great name.  Changing it to `pick_next_input_for_mode()` which makes it clear it's specialized by mode.